src/components/Image/types.ts

```diff
@@ -0,0 +1,51 @@
+import type {ImageSource} from 'expo-image';
+import type {ImageRequireSource, ImageResizeMode, ImageStyle, ImageURISource, StyleProp} from 'react-native';
+import type {OnyxEntry} from 'react-native-onyx';
+import type {Session} from '@src/types/onyx';
+
+type ExpoImageSource = ImageSource | number | ImageSource[];
+
+type ImageOnyxProps = {
+    /** Session info for the currently logged in user. */
+    session: OnyxEntry<Session>;
+};
+
+type ImageOnLoadEvent = {
+    nativeEvent: {
+        width: number;
+        height: number;
+    };
+};
+
+type ImageOwnProps = {
+    /** Styles for the Image */
+    style?: StyleProp<ImageStyle>;
+
+    /** The static asset or URI source of the image */
+    source: ExpoImageSource | Omit<ImageURISource, 'cache'> | ImageRequireSource | undefined;
+
+    /** Should an auth token be included in the image request */
+    isAuthTokenRequired?: boolean;
+
+    /** How should the image fit within its container */
+    resizeMode?: ImageResizeMode;
+
+    /** Event for when the image begins loading */
+    onLoadStart?: () => void;
```

Comment:
NAB but wouldn't `unknown` make more sense than `void` here (and for `onLoadEnd`, `onError`, `onLoad`, `onProgress`) since we just don't know or care about the return type?